src/main/java/org/opensearch/knn/training/VectorReader.java

```diff
@@ -182,10 +184,20 @@ public void onResponse(SearchResponse searchResponse) {
             int vectorsToAdd = Integer.min(maxVectorCount - collectedVectorCount, hits.length);
             List<Float[]> trainingData = new ArrayList<>();
 
-            for (int i = 0; i < vectorsToAdd; i++) {
-                trainingData.add(
-                    ((List<Number>) hits[i].getSourceAsMap().get(fieldName)).stream().map(Number::floatValue).toArray(Float[]::new)
-                );
+            for (int vector = 0; vector < vectorsToAdd; vector++) {
+                Map<String, Object> originalSourceMap = hits[vector].getSourceAsMap();
+                // The field name may be a nested field, so we need to split it and traverse the map.
+                // Example fieldName: "my_field" or "my_field.nested_field.nested_nested_field"
```

Comment:
As mentioned above,  can you check if there are any utils in OpenSearch core that perform similar functionality?

Reply:
I didn't find reusable public method from core, I'm glad to update if there are some workable methods here.